From 70de53b9d3193e77be9122d4b329492f151b3fa5 Mon Sep 17 00:00:00 2001
From: Antonio de Haro <65968177+DarthChungo@users.noreply.github.com>
Date: Mon, 8 Jun 2020 00:24:13 +0200
Subject: [PATCH] Added support for BlackWidow Chroma V2

I really doubt this will be seen by the equk, but I'm leaving this here in case anyone needs it.
---
 src/bwidow.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/src/bwidow.c b/src/bwidow.c
index b36b6d7..e0dcf64 100644
--- a/src/bwidow.c
+++ b/src/bwidow.c
@@ -28,6 +28,7 @@
 #define DEV_PID_BW_2013 0x011b
 #define DEV_PID_BW_ULT 0x010d
 #define DEV_PID_BW 0x010e
+#define DEV_PID_BW_CHROMA_V2 0x0221
 
 // USB Device
 #define DEV_INTF 2
@@ -76,10 +77,16 @@ int init () {
             //try next device
             handle = libusb_open_device_with_vid_pid(NULL, DEV_VID, DEV_PID_BW);
             if (handle == NULL) {
-            //no devices found
-                printf("ERROR - No Known Razer BlackWidow Devices Found\n");
-                libusb_exit(NULL);
-                return 1;
+                //try next device
+                handle = libusb_open_device_with_vid_pid(NULL, DEV_VID, DEV_PID_BW_CHROMA_V2);
+                if(handle == NULL){
+                    //no devices found
+                    printf("ERROR - No Known Razer BlackWidow Devices Found\n");
+                    libusb_exit(NULL);
+                    return 1;
+                } else {
+                    printf("Razer BlackWidow Chroma V2 Devide Found\n");
+                }
             } else {
                 printf("Razer BlackWidow Device Found\n");
             }
